book/website/reproducible-research/renv/renv-virtualmachine.md

```diff
@@ -88,3 +88,204 @@ alt: A screenshot showing how to export a Virtual Machine.
+Other providers can be supported by plugins.
+In particular, the Vagrant developers maintain an [official plugin](https://developer.hashicorp.com/vagrant/docs/providers/vmware/installation) for [VMWare](https://www.vmware.com/).
+For Linux users there is also a [community supported provider](https://vagrant-libvirt.github.io/vagrant-libvirt/) for [libvirt](https://libvirt.org/).
+
+```{note}
+A hypervisor is software which allows virtual machines to interact with host machines hardware at a low level.
+[This](https://www.redhat.com/en/topics/virtualization/what-is-a-hypervisor) article from Red Hat gives a good overview.
+```
+
+Vagrant environments can be packed into [boxes](https://developer.hashicorp.com/vagrant/docs/boxes).
+When using Vagrant you will most likely start from an existing box and build your environment on top of it.
+You can browse and search for public boxes [here](https://app.vagrantup.com/boxes/search).
+
+After deploying a box, Vagrant can also use [provisioners](https://developer.hashicorp.com/vagrant/docs/provisioning) to apply further configuration.
+This is useful to adapt a generic box to a specific purpose, for example by installing packages.
+Provisioning can be as simple as a shell script but can also incorporate powerful configuration management tools like [Ansible](https://docs.ansible.com/ansible/latest/index.html), [Puppet](https://puppet.com/) and [Chef](https://www.chef.io/).
+
+(rr-renv-vm-vagrant-vagrantfile)=
+### The Vagrantfile
+
+With Vagrant, users can define the configuration of a virtual machine (or group of virtual machines) in a declarative configuration language stored in a Vagrantfile.
+This configuration is written in the [Ruby](https://www.ruby-lang.org/en/) programming language.
+However, it is not necessary to know Ruby as the syntax is simple and the [documentation](https://developer.hashicorp.com/vagrant/docs/vagrantfile) explains all of the available options.
+
+```{attention}
+For Vagrant to recognise a Vagrantfile file, it must be called `Vagrantfile`.
+```
+
+Defining the virtual machines in plain text has a number of advantages over distributing full virtual machine images.
+
+- The files can be checked into version control
+- Small size makes them fast and easy to share
+- Users can reproducibly build environments
+- A single definition can potentially work across multiple hypervisors (like VirtualBox, VMWare, libvirt)
+
+In combination these qualities support Vagrant's goals of lightweight, portable and reproducible environments.
+A project can maintain its development environment alongside the source code and every contributor can use the environment with minimal barriers.
+
+(rr-renv-vm-vagrant-cli)=
+### The Vagrant CLI
+
+You will most likely use vagrant through the command line interface.
```

Comment:
Add (CLI) in brackets since this is where you first use the full term

Reply:
Fixed in [601779c](https://github.com/alan-turing-institute/the-turing-way/pull/2806/commits/601779ca1e157c061a0de720bb9d7383eee0bf0d)